src/common/Utility.js

```diff
@@ -237,6 +237,53 @@
                     return node;
                 }
             };
+        },
+        downloadData: function (format, data, id) {
+            id = id || "data";
+            var currentdate = new Date(); 
+            var timeFormat =  d3.time.format("%m-%d-%Y_%H.%M.%S");
```

Comment:
@GordonSmith You can't have a colon : in a file name (or / \ | " * ? < > ).  That's why I went with the period, I've never known it to cause an issue on modern OSes (although it did back in the day). And I wanted to go with something different than the hyphen in the date.  Any other suggestions?

I will change the order though.